spyder/widgets/explorer.py

```diff
@@ -532,7 +532,22 @@ def delete_file(self, fname, multiple, yes_to_all):
                               "<br><br>Error message:<br>%s"
                               ) % (action_str, fname, to_text_string(error)))
         return False
-
+    
+    def delete_project(self, pname):
+        """Deletes the .spyproject diretory"""
+        buttons = QMessageBox.Yes|QMessageBox.No
+        answer = QMessageBox.warning(self, _("Delete"),
+                                 _("Do you really want "
+                                   "to delete <b>%s</b>?<br><br>"
+                                   "<b>Note:</b> This action will only delete "
+                                   "the project. The files are going to be "
+                                   "preserved on disk."
+                                   ) % osp.basename(pname), buttons)
+        if answer == QMessageBox.Yes:
+            self.parent_widget.delete_project()
+            return True
+        return False

```

Comment:
In general I (probably is a personal opinion?) do not like the idea of having multiple exit points on a method (several returns...)

In this case they are so close to each other that it does not really matter but I believe having something like

is more clear
```python
result = False
if answer == QMessageBox.Yes:
    self.parent_widget.delete_project()
    result = True

return result
```

Or even

```python
result = answer == QMessageBox.Yes
if result:
    self.parent_widget.delete_project()
return result
```